File: js/src/Blockchain/Blockchain.Abstraction/Interfaces/IWorkflowActivities.ts
import { NetworkType } from '../../../Data/Entities/Networks';
import { HTLCCommitEventMessage } from '../Models/EventModels/HTLCBlockEventResposne';

export interface IWorkflowActivities {
  GetRunningWorkflowIds(workflowType: string): Promise<string[]>;

  RunEventListeningWorkflow(
    networkName: string,
    networkType: NetworkType,
    blockBatchSize: number,
    waitInterval: string
  ): Promise<void>;

  StartRefundWorkflow(swapId: string): Promise<void>;

  StartSwapWorkflow(signal: HTLCCommitEventMessage): Promise<string>;

  TerminateWorkflow(workflowId: string): Promise<void>;
}

File: csharp/src/Blockchain.Abstractions/Activities/IWorkflowActivities.cs
using Temporalio.Activities;
using Train.Solver.Blockchain.Abstractions.Models;

namespace Train.Solver.Blockchain.Abstractions.Activities;

public interface IWorkflowActivities
{
    [Activity]
    Task<List<string>> GetRunningWorkflowIdsAsync(string workflowType);

    [Activity]
    Task RunEventListeningWorkflowAsync(string networkName, uint blockBatchSize, int waitInterval);

    [Activity]
    Task StartRefundWorkflowAsync(string swapId);

    [Activity]
    Task<string> StartSwapWorkflowAsync(HTLCCommitEventMessage signal);

    [Activity]
    Task TerminateWorkflowAsync(string workflowId);
}
